### PR TITLE
#101296 Zaakdetail sync through plugins

### PR DESCRIPTION
**Pull request by Klaas-Ritense**

### CODE REVIEW

File: projects/valtimo/config/assets/core/nl.json [hunk 1]
```diff
@@ -765,6 +765,20 @@
     "selectInfoObjectType": "Kies informatieobject-type",
     "save": "Opslaan"
   },
+  "zaakdetails": {
+    "title": "Zaakdetail synchronisatie",
```
Comment (mbritense) [inline projects/valtimo/config/assets/core/nl.json]: Zaakdetail-synchronisatie

File: projects/valtimo/config/assets/core/nl.json [hunk 2]
```diff
@@ -765,6 +765,20 @@
...
+    "enabled": "Ingeschakeld",
+    "yes": "Ja",
+    "no": "Nee",
+    "objectManagementConfigurationTitle": "Object management configuratie",
```
Comment (mbritense) [inline projects/valtimo/config/assets/core/nl.json]: Object management-configuratie

File: projects/valtimo/config/assets/core/nl.json [hunk 3]
```diff
@@ -765,6 +765,20 @@
...
+    "submitButton": "Configureer zaakdetail synchronisatie",
```
Comment (mbritense) [inline projects/valtimo/config/assets/core/nl.json]: Configureer zaakdetail-synchronisatie

File: projects/valtimo/config/assets/core/nl.json [hunk 4]
```diff
@@ -765,6 +765,20 @@
...
+    "editModal": {
+      "title": "Voeg zaakdetail synchronisatie toe",
```
Comment (mbritense) [inline projects/valtimo/config/assets/core/nl.json]: Voeg zaakdetail-synchronisatie toe

File: projects/valtimo/config/assets/core/nl.json [hunk 5]
```diff
@@ -765,6 +765,20 @@
...
+      "objectManagementConfiguration": "Object management configuratie",
```
Comment (mbritense) [inline projects/valtimo/config/assets/core/nl.json]: Object management-configuratie

File: projects/valtimo/config/assets/core/nl.json [hunk 6]
```diff
@@ -765,6 +765,20 @@
...
+      "objectManagementConfigurationPlaceholder": "Selecteer een object management configuratie",
```
Comment (mbritense) [inline projects/valtimo/config/assets/core/nl.json]: Selecteer een object management-configuratie

File: projects/valtimo/zgw/src/lib/modules/index.ts [hunk 7]
```diff
@@ -18,4 +18,5 @@ export * from './documenten-api';
 export * from './objecten-api';
 export * from './contactmomenten';
 export * from './haalcentraal-brp';
+export * from './zaakdetails';
```
Comment (mbritense) [inline projects/valtimo/zgw/src/lib/modules/index.ts:21]: Don't create new zaakdetails module; move to objecten-api module

File: projects/valtimo/zgw/src/lib/modules/zaakdetails/components/document-objecten-api-sync/document-objecten-api-sync.component.ts [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+    SelectModule,
+    ValtimoCdsModalDirectiveModule,
+    CheckboxModule,
+    ReactiveFormsModule,
+    ButtonModule,
+    ModalModule,
+  ],
+})
+export class DocumentObjectenApiSyncComponent implements OnInit, OnDestroy {
+  readonly loading$ = new BehaviorSubject<boolean>(true);
```
Comment (mbritense) [inline projects/valtimo/zgw/src/lib/modules/zaakdetails/components/document-objecten-api-sync/document-objecten-api-sync.component.ts]: public readonly. Please add public/private to all class props where they are missing

File: projects/valtimo/zgw/src/lib/modules/zaakdetails/components/document-objecten-api-sync/document-objecten-api-sync.component.ts [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+  public get objectManagementConfigurationId(): AbstractControl<string> {
+    return this.formGroup.get('objectManagementConfigurationId');
+  }
+
+  public get enabled(): AbstractControl<boolean> {
+    return this.formGroup.get('enabled');
+  }
+
+  public readonly valid$ = new BehaviorSubject<boolean>(false);
+  private _subscriptions = new Subscription();
```
Comment (mbritense) [inline projects/valtimo/zgw/src/lib/modules/zaakdetails/components/document-objecten-api-sync/document-objecten-api-sync.component.ts]: private readonly

File: projects/valtimo/zgw/src/lib/modules/zaakdetails/components/document-objecten-api-sync/document-objecten-api-sync.component.ts [hunk 10]
```diff
@@ -0,0 +1,195 @@
...
+  public ngOnInit(): void {
+    this.loadDocumentenObjectenApiSync();
+  }
+
+  public ngOnDestroy(): void {
+    this._subscriptions.unsubscribe();
+  }
+
+  public loadDocumentenObjectenApiSync(): void {
+    this._subscriptions.add(
```
Comment (mbritense) [inline projects/valtimo/zgw/src/lib/modules/zaakdetails/components/document-objecten-api-sync/document-objecten-api-sync.component.ts]: No need to add this to subscriptions. It's a HTTP call observable which already quits after emitting once

File: projects/valtimo/zgw/src/lib/modules/zaakdetails/components/document-objecten-api-sync/document-objecten-api-sync.component.ts [hunk 11]
```diff
@@ -0,0 +1,195 @@
...
+          this.loading$.next(false);
+          this.configSelected(documentObjectenApiSync?.objectManagementConfigurationId);
+          this.enabled.patchValue(documentObjectenApiSync?.enabled);
+          this.documentObjectenApiSync$.next(documentObjectenApiSync);
+        })
+    );
+  }
+
+  public remove(): void {
+    this._subscriptions.add(
```
Comment (mbritense) [inline projects/valtimo/zgw/src/lib/modules/zaakdetails/components/document-objecten-api-sync/document-objecten-api-sync.component.ts]: Same, no need to add subscription

File: projects/valtimo/zgw/src/lib/modules/zaakdetails/components/document-objecten-api-sync/document-objecten-api-sync.component.ts [hunk 12]
```diff
@@ -0,0 +1,195 @@
...
+    } else {
+      this.objectManagementConfigurationId.patchValue(selectedId);
+    }
+  }
+
+  private hideModal(): void {
+    this.modalShowing$.next(false);
+  }
+
+  protected readonly CARBON_THEME = CARBON_THEME;
```
Comment (mbritense) [inline projects/valtimo/zgw/src/lib/modules/zaakdetails/components/document-objecten-api-sync/document-objecten-api-sync.component.ts]: I don't get this, CARBON_THEME is an enum but used as a value in template. Also, carbon theme should change depending on light or dark mode, can't hardcode it. Search for theme$ in code for example on how to respond to current theme changes

File: projects/valtimo/zgw/src/lib/modules/index.ts (hunk 7 above)
Comment (Klaas-Ritense) [reply projects/valtimo/zgw/src/lib/modules/index.ts:21]: Its nice to keep it the same with the BE and there we also have a new module with the same name `zaakdetails`

**Review by mbritense — APPROVED** [review]
10/10, good job

**Pull request merged**